ql/src/java/org/apache/hadoop/hive/ql/parse/AlterTableCreateBranchSpec.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.ql.parse;
+
+import com.google.common.base.MoreObjects;
+
+public class AlterTableCreateBranchSpec {
```

Comment:
can we reuse `AlterTableExecuteSpec` and define inner `CreateBranchSpec`, similar to `RollbackSpec`?

Reply:
That's what I thought before. Also I want to reuse method: `HiveIcebergStorageHandler::executeOperation` :
https://github.com/apache/hive/blob/e413b445e95e7c8beb4e43f0f6b08a98934e1c2c/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java#L645

But it  seems to be that these codes are for  syntax `alter table execute`.  Do you think we can reuse them for `alter table create branch
` or `alter table drop branch`?

Reply:
Finally, i decide not to reuse codes about `alter table execute` syntax , as i think it is more important to make code clear and readable.
 I  used `AlterTableBranchSpec ` which looks like the pattern `AlterTableExecuteSpec` , and then define inner `CreateBranchSpec` which type is `CREATE_BRANCH`. We can continue to add other brancn operation e.g `DROP_BRANCH`, `REPLACE_BRANCH`.

Please take a look again to see if it is make more sense. Thanks.

Reply:
👍